patches/tModLoader/Terraria/TileObject.cs.patch

```diff
@@ -28,6 +36,33 @@
  						flag4 = true;
  
  					if (flag4 || flag2 || flag3) {
+@@ -373,7 +_,7 @@
+ 					Tile tileSafely = Framing.GetTileSafely(num8 + num29, num9 + num28);
+ 					bool flag6 = false;
+ 					if (tileSafely.nactive()) {
+-						if (Main.tileSolid[tileSafely.type] && !Main.tileSolidTop[tileSafely.type] && !Main.tileNoAttach[tileSafely.type] && (tileData2.FlattenAnchors || tileSafely.blockType() == 0))
++						if ((anchorBottom.type & AnchorType.SolidTile) == AnchorType.SolidTile && Main.tileSolid[tileSafely.type] && !Main.tileSolidTop[tileSafely.type] && !Main.tileNoAttach[tileSafely.type] && (tileData2.FlattenAnchors || tileSafely.blockType() == 0)) // AnchorTop
```

Comment:
Do these need backporting?

Reply:
This could be backported. It doesn't technically need to since all vanilla bottom anchors are solid tile anchors from what I remember, so it doesn't affect anything.